simln-lib/src/sim_node.rs

```diff
@@ -980,16 +1189,48 @@ async fn add_htlcs(
                     }
                 }
             }
-        } else {
-            return Err(CriticalError::ChannelNotFound(scid));
-        }
+            let incoming_htlc = HtlcRef {
+                channel_id: scid,
+                index: htlc_index,
+            };
+            (incoming_htlc, next_scid)
+        };
 
-        // Once we've taken the "hop" to the destination pubkey, it becomes the source of the next outgoing htlc.
+        // Before we continue on to the next hop, we'll call any interceptors registered to get external input on the
+        // forwarding decision for this HTLC.
+        //
+        // We drop our node lock so that we can await our interceptors (which may choose to hold the HTLC for a long
+        // time) without holding our entire graph hostage.
+        drop(node_lock);
+
+        let request = InterceptRequest {
+            forwarding_node: hop.pubkey,
+            payment_hash,
+            incoming_htlc: incoming_htlc.clone(),
+            incoming_custom_records: incoming_custom_records.clone(),
+            outgoing_channel_id: next_scid,
+            incoming_amount_msat: outgoing_amount,
+            outgoing_amount_msat: outgoing_amount - hop.fee_msat,
+            incoming_expiry_height: outgoing_cltv,
+            outgoing_expiry_height: outgoing_cltv - hop.cltv_expiry_delta,
+            shutdown_listener: shutdown_signal_clone.1.clone(),
```

Comment:
If we use the top level `shutdown_signal` for intercepts and trigger on `FowardingErrors` then won't the simulation shut down any time an interceptor fails a payment? I think that it needs:
- A shutdown trigger that is used by only the interceptors
- A shutdown listener that is used by only the interceptors
- A shutdown listener that is used by the simulation

It's pretty annoying for the interceptor to have two shutdown signals, so we could look at an approach where we monitor the simulation listener and trigger the interceptor one if it's triggered (this is where child triggers would have been great).